### PR TITLE
Test and Build Hooks for release (4-2-stable)

### PR DESCRIPTION
**Pull request by d-w-moore**
Yet to do:
  - test on Ubuntu 18
  - potentially exercise more code paths by changing time of AVU collection annotation 
  - create new test overlapping indexing types on a collection [ #19 ]

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -133,7 +133,7 @@ elseif(CPACK_GENERATOR STREQUAL "DEB")
   if (CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
     set(CMAKE_SYSTEM_PROCESSOR "amd64")
   endif()
-  set(IRODS_PACKAGE_FILE_NAME_SUFFIX "_${CPACK_DEBIAN_PACKAGE_VERSION}~${IRODS_LINUX_DISTRIBUTION_VERSION_CODENAME}_${CMAKE_SYSTEM_PROCESSOR}")
+  set(IRODS_PACKAGE_FILE_NAME_SUFFIX "_${CPACK_DEBIAN_PACKAGE_VERSION}-${IRODS_LINUX_DISTRIBUTION_VERSION_CODENAME}_${CMAKE_SYSTEM_PROCESSOR}")
```
Comment (trel) [inline CMakeLists.txt:136]: pretty sure we leave these as tilde as per convention on internet...

File: irods_consortium_continuous_integration_test_hook.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+
+def main():
+    parser = optparse.OptionParser()
+    parser.add_option('--output_root_directory')
+    parser.add_option('--built_packages_root_directory')
+    parser.add_option('--indexing_engine', default='elasticsearch', help='Index/Search Platform needed for plugin test')
+    options, _ = parser.parse_args()
+
+#   output_root_directory = os.path.join(options.output_root_directory, options.indexing_engine) \
```
Comment (trel) [inline irods_consortium_continuous_integration_test_hook.py:99]: deadcode

File: packaging/document_type/postinst [hunk 3]
```diff
@@ -0,0 +1,9 @@
+if [ -f /etc/irods/service_account.config ] ; then
+  . /etc/irods/service_account.config
+else
+  IRODS_SERVICE_ACCOUNT_NAME=`stat --format "%U" /var/lib/irods`
+  IRODS_SERVICE_GROUP_NAME=`stat --format "%G" /var/lib/irods`
+fi
+
+
+#chown $IRODS_SERVICE_ACCOUNT_NAME:$IRODS_SERVICE_GROUP_NAME /var/lib/irods
```
Comment (trel) [inline packaging/document_type/postinst:9]: so we don't want this to do any healing on CentOS?

File: packaging/elasticsearch/postinst [hunk 4]
```diff
@@ -0,0 +1,9 @@
+if [ -f /etc/irods/service_account.config ] ; then
+  . /etc/irods/service_account.config
+else
+  IRODS_SERVICE_ACCOUNT_NAME=`stat --format "%U" /var/lib/irods`
+  IRODS_SERVICE_GROUP_NAME=`stat --format "%G" /var/lib/irods`
+fi
+
+
+#chown $IRODS_SERVICE_ACCOUNT_NAME:$IRODS_SERVICE_GROUP_NAME /var/lib/irods
```
Comment (trel) [inline packaging/elasticsearch/postinst:9]: so we don't want this to do any healing on CentOS?

File: CMakeLists.txt (hunk 1 above)
Comment (d-w-moore) [reply CMakeLists.txt:136]: Originally changed this due to some build or install stage giving a fatal hiccup under some conditions. Will need to investigate

File: packaging/document_type/postinst (hunk 3 above)
Comment (d-w-moore) [reply packaging/document_type/postinst:9]: this package consists only of root-owned files/directories under /usr/lib/irods, so there's no need at present. As to whether this postinst (which was created in the process of appropriating the similar [mechanism](https://github.com/irods/irods_capability_storage_tiering/blob/49011e7c70c03dfdf5bf3f303882b87d349cff3a/unified_storage_tiering.cmake#L94) within the irods storage tiering repo) technically needs to exist at all, I need to ask someone more expert in things CMake.

**Comment by d-w-moore** [comment]
tested and passing on all Ub18, Ub16, Cen7.

**Comment by trel** [comment]
excellent - please rebase and squash to single commit.

**Pull request merged**